addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/util/JAXBUtils.java

```diff
@@ -21,17 +24,31 @@
  * @author Christoph Weitkamp
  *
  */
-public class JAXBtUtils {
+public class JAXBUtils {
 
-    private static final Logger logger = LoggerFactory.getLogger(JAXBtUtils.class);
+    private static final Logger logger = LoggerFactory.getLogger(JAXBUtils.class);
 
     public static final JAXBContext JAXBCONTEXT = initJAXBContext();
 
+    public static Unmarshaller JAXBUnmarshaller;
+
     private static JAXBContext initJAXBContext() {
         try {
             return JAXBContext.newInstance(DevicelistModel.class);
         } catch (JAXBException e) {
-            logger.error("Exception creating JAXBContext: {}", e.getMessage(), e);
+            logger.error("Exception creating JAXBContext: {}", e.getLocalizedMessage(), e);
+            return null;
+        }
+    }
+
+    public static DevicelistModel buildResult(final String xml) {
+        try {
+            if (JAXBUnmarshaller == null) {
+                JAXBUnmarshaller = JAXBUtils.JAXBCONTEXT.createUnmarshaller();
+            }
+            return (DevicelistModel) JAXBUnmarshaller.unmarshal(new StringReader(xml));
+        } catch (JAXBException e) {

```

Comment:
Why 'handle' it here and return `null` the code should not continue anyway? WDYT?